src/server.py

```diff
@@ -1,59 +1,76 @@
 import sys, os
 from yaml import Loader, load
 from flask import Flask, request, Response
-from .main import main
 from werkzeug.utils import secure_filename
 from flask_swagger_ui import get_swaggerui_blueprint
+from .main import main
 
 UPLOAD_FOLDER = "/var/www/uploads"
 ALLOWED_EXTENSIONS = {"ipynb"}
-SWAGGER_URL = '/api/docs'
-API_URL = './static/swagger.yml'
+SWAGGER_URL = "/api/docs"
+API_URL = "./static/swagger.yml"
 
 app = Flask(__name__)
 app.config["UPLOAD_FOLDER"] = UPLOAD_FOLDER
 
-swagger_yml = load(open(API_URL, 'r'), Loader=Loader)
+swagger_yml = load(open(API_URL, "r", encoding="utf-8"), Loader=Loader)
 swaggerui_blueprint = get_swaggerui_blueprint(
     SWAGGER_URL,
     API_URL,
-    config = {
-        'app_name': "leakage-analysis API",
-        'spec': swagger_yml,
+    config={
+        "app_name": "leakage-analysis API",
+        "spec": swagger_yml,
     },
 )
 app.register_blueprint(swaggerui_blueprint)
 
 
+class APIError(Exception):
+    def __init__(self, message, code):
+        self.message = message
+        self.code = code
+
+
+@app.errorhandler(APIError)
+def default_error_handler(e):
+    return {"message": e.message}, e.code
+
+
 def allowed_file(filename):
     return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS
 
 
-@app.post("/upload")
-def upload():
-    file = request.files["file"]
-    if not file:
-        return "No file sent", 400
+@app.post("/analyze")
+def analyze():
+    filename = upload_file(request)
+    return analyze_file(filename)
+
+
+def upload_file(req):
+    if "file" not in req.files:
+        raise APIError("No file sent", 400)
+    file = req.files["file"]
     if not allowed_file(file.filename):
-        return "File not allowed", 400
-    filename = secure_filename(file.filename)
+        raise APIError("File not allowed", 400)
     # TODO: save in parent folder ?
```

Comment:
I would say we need to make sure to remove the #TODO before we finish the project

Reply:
Ok, je vais le mettre comme tache future dans le backlog avant.